Source/Scene/Imagery.js

```diff
@@ -85,14 +91,52 @@ define([
     };
 
     Imagery.prototype.processStateMachine = function(frameState, needGeographicProjection, priorityFunction) {
+        if (!this.imageryLayer.imageryProvider.ready || this.state === ImageryState.EMPTY) {
+            return;
+        }
+
+        var tilingScheme = this.imageryLayer.imageryProvider.tilingScheme;
+        var singleSource = !(tilingScheme instanceof ProjectedImageryTilingScheme);
+        var imageryLayer = this.imageryLayer;
+
         if (this.state === ImageryState.UNLOADED) {
             this.state = ImageryState.TRANSITIONING;
-            this.imageryLayer._requestImagery(this, priorityFunction);
+
+            if (singleSource) {
+                imageryLayer._requestImagery(this, priorityFunction);
+            } else {
+                var level = this.level;
+                var projectedIndices = tilingScheme.getProjectedTilesForNativeTile(this.x, this.y, level);
+                var projectedTilesLength = projectedIndices.length * 0.5;
+
+                this.projectedRectangles.length = projectedTilesLength;
+                this.projectedImages.length = projectedTilesLength;
+                this.projectedTextures.length = projectedTilesLength;
+
+                for (var i = 0; i < projectedTilesLength; i++) {
+                    var index = i * 2;
+                    var x = projectedIndices[index];
+                    var y = projectedIndices[index + 1];
+                    this.projectedRectangles[i] = tilingScheme.getProjectedTileProjectedRectangle(x, y, level);
+                }
+
+                if (projectedTilesLength === 0) {
+                    this.state = ImageryState.EMPTY;
```

Comment:
Shouldn't this be marked as invalid instead?

Reply:
I think if it was `INVALID` an ancestor would be rendered, but the rendered portion would be entirely transparent so we might as well just not render anything:

![img_20190116_145731914](https://user-images.githubusercontent.com/5395255/51275416-e987db00-199f-11e9-9027-809e0e4f11dd.jpg)

Reply:
What if imagery was requested past the zoom level of the imagery? It should render the ancestor there. Is there a way to tell the difference between this and the case you posted?

Reply:
Discussed offline, this happens when users don't set a max level and the terrain tile might need an `INVALID` `imagery` to upsample correctly. Just looping the state machine when the imagery isn't actually "ready" might cause problems, so I'll replace with `INVALID`.